### ISSUE TITLE
[BUG] 1.14.0-beta.rc1 Suggestions for private tags/correspondents seems to break frontend

### ISSUE THREAD

**Issue by alexgth** [paperless-ngx/paperless-ngx#3101]
### Description

Whenever paperless attempts to assign a category to an uploaded document, that the current user has no access to, the frontend breaks in multiple ways. 
1. The Suggestion sometimes shows nothing instead of "Private".
![2023-04-16_01-36](https://user-images.githubusercontent.com/65538762/232258636-5c99ef98-f152-4ded-999f-9583c7e1f9b6.png)
2. You can't always return to the Details tab while editing the document.
![2023-04-16_01-37](https://user-images.githubusercontent.com/65538762/232258669-a4a7ed1d-397b-41fb-a388-4752b78293e4.png)
3. The dropdown menus - for example to assign a correspondent - no longer collapse.
4. Assigning categories using the dropdown menus doesn't work properly. If you select anything nothing happens, but after reloading it shows the selected category.
5. The popup menus for adding new categories or deleting the file break in interesting ways.
![2023-04-16_01-39](https://user-images.githubusercontent.com/65538762/232258745-2285576a-d6ac-485f-8789-6d2007ba0d6e.png)
![2023-04-16_01-39_1](https://user-images.githubusercontent.com/65538762/232258748-a7c9204e-a1a6-446f-b89a-da369789acc8.png)

### Steps to reproduce

1. Create User A and User B
2. As User A: Create a correspondent that matches on "xyz Co." and is owned by User A permission-wise
3. As User B: Upload a file containing "xyz Co."
4. Try to assign a different correspondent or remove the empty string by clicking the 'x' on the dropdown menu and then do anything really.

### Webserver logs

```bash
None
```

### Browser logs

```bash
main.js:3 ERROR TypeError: Cannot read properties of undefined (reading 'name')
    at SX (main.js:3:595887)
    at ub (main.js:3:58787)
    at Zp (main.js:3:57564)
    at lF (main.js:3:57986)
    at Zp (main.js:3:58012)
    at lF (main.js:3:57986)
    at Zp (main.js:3:58012)
    at uF (main.js:3:64961)
    at V3 (main.js:3:58422)
    at Zp (main.js:3:58433)

This message but around 10 times per second, and not sure if that's actually related to the issue.
```

### Paperless-ngx version

1.14.0-beta.rc1

### Host OS

Ubuntu 22.04 on x86_64

### Installation method

Docker - official image

### Browser

Chrome, Vivaldi, Edge, Firefox

### Other

I've tried both the release candidate and the most recent beta version.

**Comment by shamoon** [comment]
Hmm, thanks for the detailed explanation. Are you saying only sometimes that happens but other times it (correctly) shows "Private"? I'm not sure suggestions is handled yet so I'll have to look at it, ultimately I think just some more checks will solve this. I'll take a look when I can soon

**Comment by alexgth** [comment]
Thanks for the quick response.
I think it has worked correctly twice so far. Once right after switching from release candidate to the most recent beta and the second time after trying to reproduce the bug. But once it breaks, which happens either instantly or after accessing the document in question a second time, it stays broken. Then all other documents with this kind of automatic assignment result in the same behavior.

**Comment by shamoon** [comment]
Cool, happy to have this fixed in #3103, thanks again.

**Comment by shamoon** [comment]
Closed in branch by https://github.com/paperless-ngx/paperless-ngx/commit/d457f66e8bf25d150209b789e394866cdc6e1371
